smac/optimizer/smbo.py

```diff
@@ -347,3 +342,25 @@ def _get_timebound_for_intensification(self, time_spent):
                 time_spent, (1-frac_intensify), time_left, frac_intensify))
         return time_left
 
+
+class ChallengerList(object):
+
+    def __init__(self, challengers, configuration_space):
+        self.challengers = challengers
+        self.configuration_space = configuration_space
+        self._index = 0
+        self._next_is_random = False
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._index == len(self.challengers):
+            raise StopIteration
+        elif self._next_is_random:
```

Comment:
Why do we have to change intensify for this? Maybe this was a misunderstanding: I would prefer that not every second configurations is a random configuration but we can set a parameter to change the frequency of random configurations. This would perfectly fit in this PR, since we only have to change a single line.